tmt/libraries/beakerlib.py

```diff
@@ -244,7 +248,7 @@ def fetch(self) -> None:
             # Reuse the existing metadata tree
             self.tree: fmf.Tree = library.tree
         # Fetch the library and add it to the index
-        except KeyError:
+        except (KeyError, AssertionError):
```

Comment:
Using `assert` as a flow control directive is nod a good idea. When Python is executed in optimized mode (`-O`), `assert` directives are ignored. Their purpose is to verify invariants, it’s a tool for debuggin and testing. I’m afraid you need to use proper flow control constructs like `if’.